modules/spring-config/src/main/java/org/mule/runtime/config/internal/LazyMuleArtifactContext.java

```diff
@@ -201,11 +202,25 @@ protected void prepareBeanFactory(ConfigurableListableBeanFactory beanFactory) {
     addBeanPostProcessors(beanFactory, trackingPostProcessor);
   }
 
-  private void applyLifecycle(List<String> createdComponentModels, boolean applyStartPhase) {
+  private void applyLifecycle(Map<String, Object> components, boolean applyStartPhase) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 28 to the 15 allowed. ([squid:S3776](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS3776))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7761&issues=AWnZSM1wotIXnmsU4588&open=AWnZSM1wotIXnmsU4588)

[ISSUE_KEY]: AWnZSM1wotIXnmsU4588